Fix dark theme's code block background, line number colors

m-r-mccormick opened the pull request:
When customizing `$code-background-color` in `dark.scss`, the result is a multi-color background. (see https://github.com/just-the-docs/just-the-docs/issues/1121#issuecomment-1374976843)

This makes OneDarkJekyll code block colors the same as the specified `$code-background-color`, and uses the `$default-body-color` as the line number text color, which can otherwise be invisible due to the default being black and is hard to see on a very dark code block background.

Also, `.jpg` images from just-the-docs propagate into the `_site` directory of projects created from this gem. This excludes the `assets/images` directory from being included when the gem is built by excluding any directories in the `assets` directory that start with `i`. There might be a more sustainable way to implement this but this does the trick.

mattxwang reviewed (changes requested):
Thanks for submitting a PR @m-r-mccormick! On first glance, this seems reasonable to me. I would also like to get @simonebortolin's opinion given their original development of this feature.

Two (more housekeeping) things on my end. First, could you resolve the lint errors on CI? You can view the log [in the Action run for this PR](https://github.com/just-the-docs/just-the-docs/actions/runs/3870219556/jobs/6598817318), and test locally with `npm test`. These all look autofixable.

As for

> Also, `.jpg` images from just-the-docs propagate into the `_site` directory of projects created from this gem. This excludes the `assets/images` directory from being included when the gem is built by excluding any directories in the `assets` directory that start with `i`. There might be a more sustainable way to implement this but this does the trick.

Not to be too pedantic, but could we open a separate PR for this? I think it's wholly different from the first change (and atomic / reversible commits, especially for the `.gemspec`, would be helpful). 

(as an aside, I am cleaning up the folder in #1107)

simonebortolin commented:
so, also in my opinion this PR has to be split in 2, the gemspec part is completely separate from the code background one.

for the code background, in Issue #1101 and in in PR #1100 I implemented a better separation of the scss code, making the two themes symmetrical, this PR, on the other hand, makes them even more assimilatory by having to ovverride ODJ (OneDarkJekyll).

I think your idea is absolutely reasonable, but not with the aim of overwriting the ODJ theme, but with the aim of making the code more customisable and modifiable, as soon as I have a computer at my disposal I will test the changes.

m-r-mccormick commented:
I will work on these requests, and splitting into two PRs is not too pedantic, it's reasonable and sensible. And thank you for the tips, I'm learning some new things while going through this.

m-r-mccormick commented:
I edited and ran `npm test` which passed, split excluding assets/images into a separate PR, and added a (mostly) backwards-compatible `$code-linenumber-color` variable with defaults for each dark theme.

simonebortolin commented:
Hi, these days I can't follow the discussion, in the next few days I should have more free time and finally test this PR, I hope there are no problems and/or rush to merge

mattxwang commented:
I can take point on reviewing this! I'll try to wrap up a more thorough review, but on first glance, this mostly seems good! My only concern is figuring out when `variable-exists` was introduced to SCSS - it seems like ~ 3.3, which may cause a backwards compatibility error with the `github-pages` gem.

I'll take a more in-depth look later, but if anybody else knows this immediately feel free to chime in.

mattxwang reviewed (approved):
Okay, took a look - the current version of `github-pages` allows for SASS 3.4[^1], which is good! I've also tested locally with the `github-pages` gem in our repo, and this works (i.e. compiles).

I feel confident in merging this in. I'll give it a day or so for others to chime in if they have any thoughts / if I've missed something. Thanks for your contribution!

[^1]: See: https://github.com/jekyll/jekyll-sass-converter/blob/767881d3a71846562b696353b58c84faeb81b72e/jekyll-sass-converter.gemspec#L19
